### PR TITLE
idmtest: support is-authenticate-user with domain argument

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/idmclient/107/

File: idmtest/idmtest.go [hunk 1]
```diff
@@ -218,9 +220,19 @@ func (srv *Server) user(name string) *user {
 }
 
 func (srv *Server) check(req *http.Request, cavId, cav string) ([]checkers.Caveat, error) {
-	if cav != "is-authenticated-user" {
+	if !strings.HasPrefix(cav, "is-authenticated-user") {
```
Comment (rogpeppe) [inline idmtest/idmtest.go]: or checkers.ParseCaveat?

File: idmtest/idmtest.go [hunk 2]
```diff
@@ -218,9 +220,19 @@ func (srv *Server) user(name string) *user {
...
 		return nil, errgo.Newf("unknown third party caveat %q", cav)
 	}
+	arg := strings.TrimPrefix(cav, "is-authenticated-user")
+	if len(arg) > 0 {
+		if !strings.HasPrefix(arg, " @") {
+			return nil, errgo.Newf("unknown third party caveat %q", cav)
+		}
+		domain := strings.TrimPrefix(arg, " @")
+		if !names.IsValidUserDomain(domain) {
```
Comment (rogpeppe) [inline idmtest/idmtest.go]: Shouldn't we deliver a username that has the @domain suffix requested?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/idmclient/108/

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/idmclient-merge

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/idmclient-merge

**Pull request merged**